### DISCUSSION TITLE
Dialog Popup on Close

### DISCUSSION THREAD

**Discussion by giiyms** [lepoco/wpfui#1225] [Q&A]
Hello,

I am trying to have a dialog box pop up on close so I can ask the user if they want to save their work.

I am calling contentDialogService.ShowSimpleDialogAsync from OnClosed in the main window code behind but nothing happens. It just exits without showing the dialog.

Any ideas?

**Comment by Mithicor** [comment]
Do you have your content presenter set up for the dialog?

```
<Window x:Class="MainWindow" ...>
  <Grid>
    <!-- YOUR OTHER CONTROLS AND CONTENT GOES HERE -->

    <ContentPresenter x:Name="RootContentDialog"/>
  </Grid>
</Window>
```

```
public class MainWindow
{ 
  public MainWindow()
  {
      InitializeComponent();

      contentDialogService.SetDialogHost(RootContentDialog)
  }
}
```

**Reply by giiyms** [reply]
Hello,

Yes I have a ContentDialogService.

It works fine if I click a button and show the dialog. 

But it does not work if I call the service from the OnClosed(EventArgs e) of the MainWindow.xaml.cs.

I am wondering if because it is asynchronous it fails to show the dialog before the windows closes.

I did try changed to async and await but had the same behaviour though.
